ctapipe/core/container.py

```diff
@@ -190,18 +190,27 @@ def validate(self, value):
                     f"{errorstr} Should have dimensionality {self.ndim}"
                 )
             if value.dtype != self.dtype:
-                raise FieldValidationError(
-                    f"{errorstr} Has dtype "
-                    f"{value.dtype}, should have dtype"
-                    f" {self.dtype}"
-                )
+                if np.can_cast(value, self.dtype, casting="safe"):
```

Comment:
What is the advantage here over just calling `np.astype(...,, casting="safe")` and catching the error if needed?

That should be faster for the "happy path"